src/oracle/connection/stmt.rs

```diff
@@ -59,11 +58,29 @@ impl Statement {
                 "PREPARING STMT",
             )?;
 
-            // for create statements we need to run OCIStmtPrepare2 twice
-            // c.f. https://docs.oracle.com/database/121/LNOCI/oci17msc001.htm#LNOCI17165
-            // "To reexecute a DDL statement, you must prepare the statement again using OCIStmtPrepare2()."
-            if let Some(u) = mysql.to_string().find("CREATE") {
-                if u < 10 {
+            let stmt_type =
+                Self::get_statement_type(stmt, raw_connection.env.error_handle, &mysql)?;
+            is_returning = Self::is_returning(stmt, raw_connection.env.error_handle, &mysql)?;
+
+            // c.f. https://docs.oracle.com/database/121/LNOCI/oci04sql.htm#GUID-91AF021D-9FCD-4A4D-A647-2F2AB5B448B8__CIHEHCEJ
+            // c.f. https://stackoverflow.com/a/53390359/698496
+            match stmt_type as u32 {
+                ffi::OCI_STMT_SELECT => is_select = true,
+                ffi::OCI_STMT_UPDATE
+                | ffi::OCI_STMT_DELETE
+                | ffi::OCI_STMT_INSERT
+                | ffi::OCI_STMT_BEGIN
+                | ffi::OCI_STMT_DECLARE
+                | 15
```

Comment:
At least add some comments what this magic numbers are. A better solution would be define some named constants.

Reply:
I added the reference to the SO question though